lua/neogit/popups/branch/actions.lua

```diff
@@ -22,6 +22,40 @@ local function parse_remote_branch_name(ref)
   return remote, branch_name
 end
 
+M.spin_off_branch = operation("spin_off_branch", function()
+  if #git.repo.staged.items > 0 or #git.repo.unstaged.items > 0 then
+    notif.create("Staying on current branch as there are uncommitted changes.", vim.log.levels.INFO)
+    return
+  end
+
+  local name = git.branch.create()
+
+  local upstream = git.repo.upstream.ref
```

Comment:
Prefer using the `git.branch.*` module here - in this case `git.branch.upstream_ref()` - instead of accessing repo state directly.

Reply:
You'll need to implement the input logic since I removed it from `git.branch.create()` - I'm trying to separate the interactivity layer from the library layer. Just steal the logic from here: https://github.com/NeogitOrg/neogit/blob/master/lua/neogit/popups/branch/actions.lua#L69-L73

Reply:
Couldn't find `.upstream_ref()` but assume you meant `.upstream()` - updated :)

Reply:
Added the input logic too. I'm wondering though, can I just re-use `M.create_branch()` or does that complicate things because it's a different operation?

```lua
M.create_branch = operation("create_branch", function()
  local name = input.get_user_input("branch > ")
  if not name or name == "" then
    return
  end

  name, _ = name:gsub("%s", "-")
  git.branch.create(name)
  status.refresh(true, "create_branch")
end)
```

Reply:
Definitely don't re-use it. While these functions are all co-located, they should be thought of in isolation. `DRY` as in `Definitely repeat yourself` :P

Reply:
Make sense 👍🏼 😄